### ISSUE TITLE
Windows: Installer option for JavaSoft registry entries missing

### ISSUE THREAD

**Issue by jpstotz** [adoptium/installer#300]
The latest version AdoptOpenJDK 11 version `OpenJDK11U-jdk_x64_windows_hotspot_11.0.10_9.msi` misses the option to create the JavaSoft registry entries.

Those entries are required by a lot of programs to find an existing JRE/JDK installation (e.g. Lauch4j).

**Comment by douph1** [comment]
Sorry I don't see it missing but for some language the translation item name and desc seems not to work anymore.

Translation KO :
>de-de 1031      cp1252
>ja-jp 1041	cp932
>zh-cn 2052	cp936
>zh-tw 1028	cp950 

What language are you using ? Can you share a screenshot ?

**Comment by jpstotz** [comment]
@douph1 My Windows system is using German, hence it should be de-de.

The installer dialog shows the following options on my system:

![installer](https://user-images.githubusercontent.com/924885/112114523-72ba7e00-8bb8-11eb-91b1-9497e4e0bc56.png)

**Comment by jpstotz** [comment]
I found the reason for this defect:

I have also installed AdoptOpenJDK 15 JDK for experiments. 
This version has set the registry value `HKLM\SOFTWARE\JavaSoft\JDK\CurrentVersion` to `15`.

If the installer detects this value then the option `JavaSoft (Oracle) registry keys` is not shown. If I rename this key the JavaSoft option is visible again.

This behavior does not make any sense to me. The JavaSoft registry values exist so that it is possible to detect all installed Java versions and select the best matching JDK/JRE. Therefore the JavaSoft option should be selectable no matter what other JREs/JDKs are already installed.

**Comment by douph1** [comment]
see https://github.com/AdoptOpenJDK/openjdk-installer/blob/master/wix/Main.hotspot.wxs.template
>        
>        <!-- As Oracle, don't override if current value is already bigger ( newer java ): "a value that is the highest installed version on the system" 


Oracle : 
>This key contains the string CurrentVersion, with a value **that is the highest installed version on the system**.

If we allow you to override CurrentVersion from JDK 15 with value from JDK11 we will not respect this.

**Comment by jpstotz** [comment]
@douph1 The option `JavaSoft (Oracle) registry keys` does not only write this value. In my opinion the registry value I mentioned is totally useless (therefore it is not a good idea to use it to determine if the installer option is shown or not), the more important registry values for a JDK 11 are:

```
[HKEY_LOCAL_MACHINE\SOFTWARE\JavaSoft\Java Development Kit\11]
"JavaHome"="C:\\Program Files\\AdoptOpenJDK\\jdk-11.0.10.9-hotspot"

[HKEY_LOCAL_MACHINE\SOFTWARE\JavaSoft\JDK\11]
"JavaHome"="C:\\Program Files\\AdoptOpenJDK\\jdk-11.0.10.9-hotspot\\"

[HKEY_LOCAL_MACHINE\SOFTWARE\JavaSoft\JDK\11.0.10.9]
"JavaHome"="C:\\Program Files\\AdoptOpenJDK\\jdk-11.0.10.9-hotspot\\"
```
These are the values that really matters because they can be used to find an installed JDK.
And these values do not exist if you install AdoptOpenJDK 15, thus nothing is overwritten. 

If the CurrentVersion entry is found you can just skip it but please allow the installer to write the other entries.

**Comment by douph1** [comment]
Are you suggesting to have two installer options ?
- One for CurrentVersion as now which is not display to not allow overriding bigger CurrentVersion already installed
- another installer option to allow to write the other JavaSoft key ?

How two handle backward compat ?

**Comment by jpstotz** [comment]
That is not necessary. But if you want to mimic the Oracle behavior you can always show the JavaSoft option and if it is selected but `HKLM\SOFTWARE\JavaSoft\JDK\CurrentVersion` is set to a higher value just skip writing this value. 

That would the the obvious logical way to handle such a situation in my opinion. 

But please do never hide the JavaSoft installer  option.

**Comment by douph1** [comment]
Seems it was fix by #300. Feel free to reopen if needed
